Onboarding - Add business extension recommendations

justinshreve opened the pull request:
Fixes #2518.

This PR adds the Mailchimp and Facebook extensions to the business tab of the profile wizard.

<img width="849" alt="Screen Shot 2019-07-01 at 4 54 59 PM" src="https://user-images.githubusercontent.com/689165/60466647-3d227400-9c22-11e9-9046-54bc85cad664.png">

Note: I noticed certain values for `other_platform` are causing save endpoints to fail with a 400, even though they should be valid. This is not addressed in this PR. I can take a look at this tomorrow and fix in a follow-up PR.

### Detailed test instructions:

* Go to `/wp-admin/admin.php?page=wc-admin#/?step=business-details`
* Fill out the form and see that the extension recommendations display. Test the toggles, and saving your profile data. You can request `GET /wp-json/wc-admin/v1/onboarding/profile` to make sure the extensions saved.

joshuatf commented on client/dashboard/profile-wizard/steps/business-details.js:
```diff
@@ -83,6 +92,85 @@ class BusinessDetails extends Component {
+		return (
+			<p>
+				{ sprintf(
+					__( 'The following plugins will be installed for free: %s' ),
```
👍

joshuatf commented on client/dashboard/profile-wizard/style.scss:
```diff
@@ -213,6 +213,112 @@
+	}
+
+	.woocommerce-profile-wizard__tracking {
+		.components-form-toggle {
```
My toggles look a little off-centered vertically:

<img width="59" alt="Screen Shot 2019-07-02 at 10 20 41 AM" src="https://user-images.githubusercontent.com/10561050/60477859-31cb5900-9cb3-11e9-9d5a-ecf90c863cb5.png">

joshuatf commented on includes/api/class-wc-admin-rest-onboarding-profile-controller.php:
```diff
@@ -283,15 +283,27 @@ public static function get_profile_properties() {
 				),
 			),
-			'theme'           => array(
+			'business_extensions' => array(
```
We'll need a couple quick changes in  `tests/api/onboarding-profile.php` to get tests ✅

justinshreve replied:
Done.

justinshreve commented:
I've fixed the phpunit test.

@joshuatf I can't seem to replicate the off-center toggles. Your toggles look a bit bigger than mine, which could affect the positioning, so I did add a max height/width. Does that help? If not, could you see what CSS is causing that? I'm not having issues in Chrome or Safari.

joshuatf commented:
@justinshreve Odd, I'm seeing it in all browsers.  I wonder if there's another plugin altering the behavior here.

>Your toggles look a bit bigger than mine

I think I may have been zoomed in.  Mine appear the same size as the ones in your screenshot.

The span here appears a bit off:

<img width="470" alt="Screen Shot 2019-07-03 at 4 13 54 PM" src="https://user-images.githubusercontent.com/10561050/60575259-d336d580-9dad-11e9-8854-51b0bdad1f37.png">

But adding display `block` or `inline-block` to the toggle seems to position that span at the top:

`.components-form-toggle {
    display: inline-block;
}`

<img width="49" alt="Screen Shot 2019-07-03 at 4 14 35 PM" src="https://user-images.githubusercontent.com/10561050/60575367-f8c3df00-9dad-11e9-982c-8a0be83dfc25.png">

justinshreve commented:
@joshuatf Thanks -- I agree there must be some plugin difference (or maybe Gutenberg difference) because I already had that class coming in from wp components, but I've added that class in our CSS as well.

I've also adjusted when the extensions display so that the extensions don't disappear/reappear as you select from the selling dropdown (which happened after rebase now that we set a default onFocus).

justinshreve commented:
> It feels like these styles should be included in @wordpress/components, but I guess that would create duplication between the WP included styles when updating. Is there some discussion around this already that I can read up on?

Good point... I agree. I did a quick search but don't see any discussion around that.
